include/odp/api/abi-default/atomic.h

```diff
@@ -20,6 +20,22 @@ extern "C" {
 #include <odp/api/std_types.h>
 #include <odp/api/align.h>
 
+/**
+ * @internal
+ * 128-bit integer structure.
+ */
+struct odp_u128_s{
+	union {
+		unsigned __int128 u128;
+		uint64_t u64[2];
+		uint32_t u32[4];
+		uint16_t u16[8];
+		uint8_t  u8[16];
+	};
+} ODP_ALIGNED(16);
+
+typedef struct odp_u128_s odp_u128_t;
```

Comment:
This type should be defined in API spec file (std_types.h) since application uses it's fields directly. __int128 is problematic to include in a spec file as it's not a standard C type. So, may be we could just leave it away (at least initially). The main use case (atomic CAS of two 64bit pointers) would be still supported.

Reply:
I have removed __int128 from the structure. 
The structure definition is moved into include/odp/api/spec/std_types.h.

I am not clear whether the definition should go into include/odp/api/abi-default/std_types.h or include/odp/api/spec/std_types.h ?

Reply:
Spec is the correct place since application can refer those definitions directly. ABI files are implementation specific definitions and application must not refer those directly, but use API spec defintions (e.g. use atomic function calls instead of pointing directly into odp_atomic_u128_t struct fields).